### ISSUE TITLE
Tensorflow - optional installation and running

### ISSUE THREAD

**Issue by ewallace** [fish-quant/big-fish-examples#12]
Hello,
Is tensorflow & tensorflow addons needed _only_ for the U-net cell segmentation in notebook [4 - Segment nuclei and cells.ipynb](notebooks/4%20-%20Segment%20nuclei%20and%20cells.ipynb)?

I think it would be helpful to flag that to the user. This was a problem for me because I failed to setup my environment to have tensorflow working.

So in notebook 4 I would find it helpful if the U-net section were edited to say something like:

> ## U-net based model 

> U-net is a neural-network based model for image segmentation, see [wikipedia on U-net](https://en.wikipedia.org/wiki/U-Net). We implement a U-net based model to directly segment nucleus. It could be particularly useful to discriminate adjacent nuclei. 

> The big-fish implementation of U-net requires tensorflow and tensorflow_addons packages. Skip this step if you don't have tensorflow installed or if you don't need to use this algorithm. Check that you have tensorflow installed first:

```
import tensorflow
import tensorflow addons
```

That would clarify the situation and make it easier to troubleshoot the installation if that's the problem. Then the notebook could move on to: 

> Segmentation is performed in two steps:
(and so on...)

It would also be helpful if the file saving output section was separate for the thresholding and U-net masks, if I understand that right?

## Installation problems and package versions

I didn't try very hard to install tensorflow_addons. Basically I was running from python 3.9 and big-fish 0.6.2. I successfully installed tensorflow 2.4.1, then pip would only install tensorflow_addons version 0.16.1 or above. The kernel for notebook 4 ran the `# load pretrained model` cell, crashed at the `# instance segmentation` cell. 

My installation commands:

```
conda create -n bigfish_env python=3.9
pip install big-fish
conda install jupyter
conda install --channel conda-forge tensorflow  
pip install --channel conda-forge tensorflow_addons==0.16.1 
```

I'm sure this is a stupid mistake by me or comes from me using python 3.9, I can check in more detail if that's helpful to you?

**Comment by Henley13** [comment]
Hello @ewallace,

Oh it is not a mistake from you, it's just that I pinned TensorFlow version to 2.3.0 and it does not support python >= 3.9. I will need to unpin the deep learning dependencies. As you suggest, I should could also clarify that these dependencies are optional to use the majority of Big-FISH.

**Comment by Henley13** [comment]
Concerning the saving output section, it only reuses the thresholding and watershed techniques. So no deep learning is needed to run the end of the notebook.
